File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-confirmation-processes-5-4/ClearingStatus.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ClearingStatus : NotificationMessage
    {
        public ClearingStatus(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode clearingStatusItemNode = xmlNode.SelectSingleNode("clearingStatusItem");
            
            if (clearingStatusItemNode != null)
            {
                if (clearingStatusItemNode.Attributes["href"] != null || clearingStatusItemNode.Attributes["id"] != null) 
                {
                    if (clearingStatusItemNode.Attributes["id"] != null) 
                    {
                        clearingStatusItemIDRef_ = clearingStatusItemNode.Attributes["id"].Value;
                        ClearingStatusItem ob = new ClearingStatusItem(clearingStatusItemNode);
                        IDManager.SetID(clearingStatusItemIDRef_, ob);
                    }
                    else if (clearingStatusItemNode.Attributes["href"] != null)
                    {
                        clearingStatusItemIDRef_ = clearingStatusItemNode.Attributes["href"].Value;
                    }
                    else
                    {
                        clearingStatusItem_ = new ClearingStatusItem(clearingStatusItemNode);
                    }
                }
                else
                {
                    clearingStatusItem_ = new ClearingStatusItem(clearingStatusItemNode);
                }
            }
            
        
            XmlNodeList partyNodeList = xmlNode.SelectNodes("party");
            
            if (partyNodeList != null)
            {
                this.party_ = new List<Party>();
                foreach (XmlNode item in partyNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            partyIDRef_ = item.Attributes["id"].Value;
                            party_.Add(new Party(item));
                            IDManager.SetID(partyIDRef_, party_[party_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            partyIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        party_.Add(new Party(item));
                        }
                    }
                    else
                    {
                        party_.Add(new Party(item));
                    }
                }
            }
            
        
            XmlNodeList accountNodeList = xmlNode.SelectNodes("account");
            
            if (accountNodeList != null)
            {
                this.account_ = new List<Account>();
                foreach (XmlNode item in accountNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            accountIDRef_ = item.Attributes["id"].Value;
                            account_.Add(new Account(item));
                            IDManager.SetID(accountIDRef_, account_[account_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            accountIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        account_.Add(new Account(item));
                        }
                    }
                    else
                    {
                        account_.Add(new Account(item));
                    }
                }
            }
            
        
        }
        
    
        #region ClearingStatusItem_
        private ClearingStatusItem clearingStatusItem_;
        public ClearingStatusItem ClearingStatusItem_
        {
            get
            {
                if (this.clearingStatusItem_ != null)
                {
                    return this.clearingStatusItem_; 
                }
                else if (this.clearingStatusItemIDRef_ != null)
                {
                    clearingStatusItem_ = IDManager.getID(clearingStatusItemIDRef_) as ClearingStatusItem;
                    return this.clearingStatusItem_; 
                }
                else
                {
                      return this.clearingStatusItem_; 
                }
            }
            set
            {
                if (this.clearingStatusItem_ != value)
                {
                    this.clearingStatusItem_ = value;
                }
            }
        }
        #endregion
        
        public string clearingStatusItemIDRef_ { get; set; }
        #region Party_
        private List<Party> party_;
        public List<Party> Party_
        {
            get
            {
                if (this.party_ != null)
                {
                    return this.party_; 
                }
                else if (this.partyIDRef_ != null)
                {
                    return this.party_; 
                }
                else
                {
                      return this.party_; 
                }
            }
            set
            {
                if (this.party_ != value)
                {
                    this.party_ = value;
                }
            }
        }
        #endregion
        
        public string partyIDRef_ { get; set; }
        #region Account_
        private List<Account> account_;
        public List<Account> Account_
        {
            get
            {
                if (this.account_ != null)
                {
                    return this.account_; 
                }
                else if (this.accountIDRef_ != null)
                {
                    return this.account_; 
                }
                else
                {
                      return this.account_; 
                }
            }
            set
            {
                if (this.account_ != value)
                {
                    this.account_ = value;
                }
            }
        }
        #endregion
        
        public string accountIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-confirmation-processes-5-4/ClearingStatus.cpp
// ClearingStatus.cpp 
#include "ClearingStatus.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ClearingStatus::ClearingStatus(TiXmlNode* xmlNode)
: NotificationMessage(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //clearingStatusItemNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* clearingStatusItemNode = xmlNode->FirstChildElement("clearingStatusItem");

   if(clearingStatusItemNode){clearingStatusItemIsNull_ = false;}
   else{clearingStatusItemIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- clearingStatusItemNode , address : " << clearingStatusItemNode << std::endl;
   #endif
   if(clearingStatusItemNode)
   {
      if(clearingStatusItemNode->Attribute("href") || clearingStatusItemNode->Attribute("id"))
      {
          if(clearingStatusItemNode->Attribute("id"))
          {
             clearingStatusItemIDRef_ = clearingStatusItemNode->Attribute("id");
             clearingStatusItem_ = boost::shared_ptr<ClearingStatusItem>(new ClearingStatusItem(clearingStatusItemNode));
             clearingStatusItem_->setID(clearingStatusItemIDRef_);
             IDManager::instance().setID(clearingStatusItemIDRef_,clearingStatusItem_);
          }
          else if(clearingStatusItemNode->Attribute("href")) { clearingStatusItemIDRef_ = clearingStatusItemNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { clearingStatusItem_ = boost::shared_ptr<ClearingStatusItem>(new ClearingStatusItem(clearingStatusItemNode));}
   }

   //partyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* partyNode = xmlNode->FirstChildElement("party");

   if(partyNode){partyIsNull_ = false;}
   else{partyIsNull_ = true;}

   if(partyNode)
   {
      for(partyNode; partyNode; partyNode = partyNode->NextSiblingElement("party")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyNode , address : " << partyNode << std::endl;
          #endif
          if(partyNode->Attribute("href") || partyNode->Attribute("id"))
          {
              if(partyNode->Attribute("id"))
              {
                  partyIDRef_ = partyNode->Attribute("id");
                  party_.push_back(boost::shared_ptr<Party>(new Party(partyNode)));
                  party_.back()->setID(partyIDRef_);
                  IDManager::instance().setID(partyIDRef_, party_.back());
              }
              else if(partyNode->Attribute("href")) { partyIDRef_ = partyNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { party_.push_back(boost::shared_ptr<Party>(new Party(partyNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyNode , address : " << partyNode << std::endl;
       #endif
   }

   //accountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* accountNode = xmlNode->FirstChildElement("account");

   if(accountNode){accountIsNull_ = false;}
   else{accountIsNull_ = true;}

   if(accountNode)
   {
      for(accountNode; accountNode; accountNode = accountNode->NextSiblingElement("account")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accountNode , address : " << accountNode << std::endl;
          #endif
          if(accountNode->Attribute("href") || accountNode->Attribute("id"))
          {
              if(accountNode->Attribute("id"))
              {
                  accountIDRef_ = accountNode->Attribute("id");
                  account_.push_back(boost::shared_ptr<Account>(new Account(accountNode)));
                  account_.back()->setID(accountIDRef_);
                  IDManager::instance().setID(accountIDRef_, account_.back());
              }
              else if(accountNode->Attribute("href")) { accountIDRef_ = accountNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { account_.push_back(boost::shared_ptr<Account>(new Account(accountNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accountNode , address : " << accountNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<ClearingStatusItem> ClearingStatus::getClearingStatusItem()
{
   if(!this->clearingStatusItemIsNull_){
        if(clearingStatusItemIDRef_ != ""){
             return boost::shared_static_cast<ClearingStatusItem>(IDManager::instance().getID(clearingStatusItemIDRef_));
        }else{
             return this->clearingStatusItem_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ClearingStatusItem>();
   }
}
std::vector<boost::shared_ptr<Party>> ClearingStatus::getParty()
{
   if(!this->partyIsNull_){
        if(partyIDRef_ != ""){
             return this->party_;
        }else{
             return this->party_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Party>>();
   }
}
std::vector<boost::shared_ptr<Account>> ClearingStatus::getAccount()
{
   if(!this->accountIsNull_){
        if(accountIDRef_ != ""){
             return this->account_;
        }else{
             return this->account_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Account>>();
   }
}
}

